components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/listener/IdentityStoreEventListener.java

```diff
@@ -251,9 +251,11 @@ public boolean doPostGetUserClaimValues(String userName, String[] claims, String
             return true;
         }
         // data found, add the values for security questions and identity claims
+        String value;
         for (String claim : claims) {
-            if (identityDTO.getUserIdentityDataMap().containsKey(claim)) {
-                claimMap.put(claim, identityDTO.getUserIdentityDataMap().get(claim));
+            if (identityDTO.getUserIdentityDataMap().containsKey(claim)
+                    && (value = identityDTO.getUserIdentityDataMap().get(claim)) != null && value.trim().length() > 0) {
```

Comment:
Can we use `StringUtils.isNotBlank()` here? WDYT?

Reply:
I guess StringUtils.isNotBlank() would also work but what's the issue with this?

Reply:
We encourage to use `StringUtils`, `MapUtils`, `CollectionUtils` etc. for null checks and not empty checks since its a good practice.

Reply:
Ok done. FYI https://www.google.com/search?safe=active&sxsrf=ACYBGNR0E8U5YVd1eF_12fzU-GW4jRuwYA%3A1581657209267&ei=eSxGXoP7D6-W4-EPgNCnmA4&q=if+%28value+%3D+null+%26%26+value.trim%28%29.length%28%29+%3E+0%29+wso2+github&oq=if+%28value+%3D+null+%26%26+value.trim%28%29.length%28%29+%3E+0%29+wso2+github&gs_l=psy-ab.3..35i39.15040.15814..16687...0.0..0.124.388.2j2......0....1..gws-wiz.1GhDqiL8140&ved=0ahUKEwjDm_WgpNDnAhUvyzgGHQDoCeMQ4dUDCAs&uact=5

Reply:
Yes, we have used it before. Recently we have started a special effort to improve code quality. As a part of this effort, we have initiated these improvements in PRs.